### PR TITLE
add order by behaviour with array similar to get method

### PR DESCRIPTION
**Pull request by Mohit21GoJs**
fix issue #[4438](https://github.com/lodash/lodash/issues/4438)

### CODE REVIEW

**Comment by jdalton** [comment]
Hi @Mohit21GoJs!

Could you rebase this PR without the package-lock.json update?

**Comment by Mohit21GoJs** [comment]
> Hi @Mohit21GoJs!
> 
> Could you rebase this PR without the package-lock.json update?

Updated @jdalton , please review.

File: .internal/baseOrderBy.js [hunk 1]
```diff
@@ -15,7 +16,12 @@ import isArrayLike from '../isArrayLike.js'
 function baseOrderBy(collection, iteratees, orders) {
   let criteriaIndex = -1
   let eachIndex = -1
-  iteratees = iteratees.length ? iteratees : [(value) => value]
+  iteratees = iteratees.length ? iteratees.map((iteratee) => {
+    if (Array.isArray(iteratee)) {
+      return (value) => baseGet(value, iteratee)
+    }
+    return iteratee
+  }) : [(value) => value]
```
Comment (jdalton) [inline .internal/baseOrderBy.js]: We have a method called `identity` that can me used for `(value) => value`.

Comment (Mohit21GoJs) [reply .internal/baseOrderBy.js]: @jdalton the existing identity was defined in ../test/util.js, hence i defined a identity function in this file as there is no other util file, let me know if it makes sense.

**Comment by jdalton** [comment]
Ah cool. Thanks @Mohit21GoJs!

**Comment by falsyvalues** [comment]
@jdalton Shall we backport this to `v4.17.15-post`?

**Comment by jdalton** [comment]
@falsyvalues That would be a good idea!

**Pull request merged**